cai_causal_graph/time_series_causal_graph.py

```diff
@@ -0,0 +1,880 @@
+
+        # populate the metadata for each node
+        if meta is not None:
+            meta.update({'time_lag': time_lag, 'variable_name': variable_name})
+        else:
+            meta = {'time_lag': time_lag, 'variable_name': variable_name}
+
+        # populate the metadata for each node
+        super().__init__(identifier, meta, variable_type)
+
+    @property
+    def time_lag(self) -> int:
+        """Return the time lag of the node from the metadata."""
+        return self.meta.get('time_lag', 0)
+
+    @property
+    def variable_name(self) -> Optional[str]:
+        """Return the variable name of the node from the metadata."""
+        return self.meta.get('variable_name', None)
+
+    def __repr__(self) -> str:
+        return super().__repr__().replace('Node', 'TimeSeriesNode')
+
+
+class TimeSeriesCausalGraph(CausalGraph):
+    """
+    A causal graph for time series data.
+
+    The node in a time series causal graph will have additional metadata that
+    gives the time information of the node together with the variable name.
+    The two additional metadata are:
+    - 'time_lag': the time difference with respect the reference time 0
+    - 'variable_name': the name of the variable (without the lag information).
+    """
+
+    def __init__(
+        self,
+        input_list: Optional[List[NodeLike]] = None,
+        output_list: Optional[List[NodeLike]] = None,
+        fully_connected: bool = True,
+    ):
+        """
+        Initialize the time series causal graph.
+
+        :param input_list: list of input nodes. Default is None.
+        :param output_list: list of output nodes. Default is None.
+        :param fully_connected: if True, the graph will be fully connected.
+            Default is True.
+
+        Example usage:
+            >>> from cai_causal_graph import CausalGraph, TimeSeriesCausalGraph
+            >>>
+            >>>
+            >>> # How to initialize a TimeSeriesCausalGraph directly
+            >>> ts_cg = TimeSeriesCausalGraph()
+            >>> ts_cg.add_edge('X1 lag(n=1)', 'X1', edge_type=EDGE_T.DIRECTED_EDGE)
+            >>> ts_cg.add_edge('X2 lag(n=1)', 'X2', edge_type=EDGE_T.DIRECTED_EDGE)
+            >>> ts_cg.add_edge('X1', 'X3', edge_type=EDGE_T.DIRECTED_EDGE)
+            >>>
+            >>>
+            >>> # How to initialize a TimeSeriesCausalGraph from a CausalGraph
+            >>> cg = CausalGraph()
+            >>> cg.add_edge('X1 lag(n=1)', 'X1', edge_type=EDGE_T.DIRECTED_EDGE)
+            >>> cg.add_edge('X2 lag(n=1)', 'X2', edge_type=EDGE_T.DIRECTED_EDGE)
+            >>> cg.add_edge('X1', 'X3', edge_type=EDGE_T.DIRECTED_EDGE)
+            >>>
+            >>> # The time series causal graph will have the same nodes and edges as the causal graph
+            >>> # but it is aware of the time information so 'X1 lag(n=1)' and 'X1' represent the same
+            >>> # variable but at different times.
+            >>> ts_cg = TimeSeriesCausalGraph.from_causal_graph(cg)
```

Comment:
Shouldn't the example usage normally be above the ":param" parameter docstrings? Would vote for swapping